lib/phoenix_live_view/html_formatter.ex

```diff
@@ -552,7 +554,42 @@ defmodule Phoenix.LiveView.HTMLFormatter do
     [{:tag_block, name, attrs, block, %{mode: mode}} | list]
   end
 
-  defp may_set_preserve(buffer, _text), do: buffer
+  @non_ws_preserving_elements ["button"]
+
+  defp may_set_preserve_on_block(buffer, _text), do: buffer
+
+  defp may_set_preserve_on_text([{:text, text, meta}], :inline, tag_name)
+       when tag_name not in @non_ws_preserving_elements do
+    {mode, text} =
+      if meta.newlines == 0 and whitespace_around?(text) do
+        {:preserve, cleanup_extra_spaces(text)}
```

Comment:
```suggestion
        text =
          text
          |> cleanup_extra_spaces_leading()
          |> cleanup_extra_spaces_trailing()

        {:preserve, text}
```

_See the other suggestion_